public/externalLibs/graphics/webGLrune.js

```diff
@@ -350,6 +350,9 @@ function generateFlattenedRuneList(rune) {
     }
   }
   function helper(rune, color) {
+    if (!isRune(rune)) {
+        throw 'primitive rune function expects rune, received ' + rune
```

Comment:
If I'm not wrong, I think string conversion for arrays omit the brackets, "" + [1, 2] --> "1, 2".

We probably want to JSON.stringify the argument, but this would re-introduce the long error messages.

Ideally, we can try to override the .toJSON member in all runes (and generated runes made by transformations) so that JSON.stringify gives a nicer output. e.g.

```
let blah = heart;
blah.toJSON = () => '{ [rune] }';
console.log(JSON.stringify([blah])); // prints ["{ [rune] }"]
```